packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowContainerTabs.ts

```diff
@@ -12,114 +17,352 @@ import {
+        ifFeatureFlagsEnabled: ['IS_WORKFLOW_ENABLED'],
+        ifObjectsDontExist: [],
+        ifRelationsDontExist: [],
+      },
     },
     {
       id: 'workflowVersion',
       title: 'Flow',
       Icon: IconSettings,
-      hide: !isWorkflowVersion,
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        ifFeatureFlagsEnabled: ['IS_WORKFLOW_ENABLED'],
+        ifObjectsDontExist: [],
+        ifRelationsDontExist: [],
+      },
     },
     {
       id: 'workflowRunOutput',
       title: 'Output',
       Icon: IconPrinter,
-      hide: !isWorkflowRun,
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        ifFeatureFlagsEnabled: ['IS_WORKFLOW_ENABLED'],
+        ifObjectsDontExist: [],
+        ifRelationsDontExist: [],
+      },
     },
     {
       id: 'workflowRunFlow',
       title: 'Flow',
       Icon: IconSettings,
-      hide: !isWorkflowRun,
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        ifFeatureFlagsEnabled: ['IS_WORKFLOW_ENABLED'],
+        ifObjectsDontExist: [],
+        ifRelationsDontExist: [],
+      },
     },
   ];
+
+  // Process tabs and apply visibility logic
+  return tabDefinitions.map(({ id, title, Icon, hide }) => {
+    // Special handling for fields tab
+    if (id === 'fields') {
+      return {
+        id,
+        title,
+        Icon,
+        hide: !(isMobile || isInRightDrawer), // Show only on mobile or in right drawer
+      };
+    }
+
+    const baseHide =
+      (hide.onMobile && isMobile) ||
+      (hide.onDesktop && !isMobile) ||
+      (hide.inRightDrawer && isInRightDrawer) ||
+      (hide.ifFeatureFlagsEnabled.length > 0 &&
+        !hide.ifFeatureFlagsEnabled.every((flag) => {
+          if (flag === 'IS_WORKFLOW_ENABLED') return isWorkflowEnabled;
+          return false;
+        }));
+
+    const objectsDontExist =
+      hide.ifObjectsDontExist.length > 0 &&
+      !hide.ifObjectsDontExist.every((obj) => {
+        switch (obj) {
+          case 'Note':
+            return isNotesObjectActive;
+          case 'Task':
+            return isTasksObjectActive;
+          case 'Attachment':
+            return isAttachmentsObjectActive;
+          default:
+            return objectMetadataItems.some(
+              (item) => item.nameSingular === obj && item.isActive,
+            );
+        }
+      });
+
+    const relationsDontExist =
+      hide.ifRelationsDontExist.length > 0 &&
+      !hide.ifRelationsDontExist.every((rel) =>
+        objectMetadataItem.fields.some(
+          (field) =>
+            field.type === FieldMetadataType.Relation &&
+            field.name === rel &&
+            field.isActive,
+        ),
+      );
+
+    let isHidden =
+      loading || baseHide || objectsDontExist || relationsDontExist;
+
+    // Special case handling from original code
+    if (!isHidden) {
```

Comment:
Hey, this is a good step forward but I was hoping you'd get rid of most of this code!
The goal is not to have to create 
```      
case 'Note':
            return isNotesObjectActive;
      ```
      
   But instead directly pass `CoreObjectNameSingular.Note` etc